### PR TITLE
Feature/113 scrublet support

### LINKED ISSUE #113: Remove doublets (Scrublet)
#not so nice

```
import scanpy as sc
import numpy as np
import matplotlib as mpl
import matplotlib.pyplot as plt
import pandas as pd
adata = sc.read_h5ad("FCA_10x_All.SC__FILE_CONCATENATOR.h5ad")
adata_dr = sc.read_h5ad("FCA_10x_All.SC__SCANPY__DIM_REDUCTION_UMAP.h5ad")

#take variable genes from adata_dr
variable_genes = list(adata_dr.var.index[adata_dr.var['highly_variable']])
print(len(variable_genes))
doublet_cells_all = []

batches = list(set(adata.obs['batch']))
adata_objects_raw = {}
for batch in batches:
    adata_0 = adata[adata.obs['batch']==batch,:].copy()
    adata_dr_0 = adata_dr[adata_dr.obs['batch']==batch,:].copy()
    adata_0_var = adata_0.X[:,np.array(adata_dr_0.var['highly_variable'])] #use adata_dr_0 te keep genes at the same position, output is a matrix
    scrub = scr.Scrublet(adata_0_var)
    adata_0.obs['doublet_scores'], adata_0.obs['predicted_doublets'] = scrub.scrub_doublets(synthetic_doublet_umi_subsampling=0.5,n_prin_comps=50)
    scrub.plot_histogram()
    adata_objects_raw[batch] = adata_0
    #take doublet cells based on expected doublets based on number of cells (10x Chromium)
    #expected_doublets = np.int(len(adata_0)/100000*len(adata_0)) (updated see comments)
    cells_recovered = len(adata_0)
    doublet_rate = 0.0008 * cells_recovered + 0.0527
    expected_doublets = np.int(doublet_rate/100 * cells_recovered)


    doublet_cells = adata_0.obs['doublet_scores'].sort_values(ascending=False).head(n=expected_doublets).index
    doublet_cells_all = doublet_cells_all + list(doublet_cells)

adata_dr.obs['man_doublets'] = False
adata_dr.obs.loc[doublet_cells_all,'man_doublets'] = True
sc.pl.tsne(adata_dr, color='man_doublets',size=10,color_map='plasma_r')
```

### PR DESCRIPTION
**Pull request by dweemx**
- Closes #113 
- Various cleanups and refactorings

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -43,3 +43,7 @@
 [submodule "src/popscle"]
 	path = src/popscle
 	url = https://github.com/vib-singlecell-nf/popscle.git
+[submodule "src/scrublet"]
+	path = src/scrublet
+	url = git@github.com:vib-singlecell-nf/scrublet.git
```
Comment (KrisDavie) [inline .gitmodules:48]: For consistency we should move all of these to ssh based rather than https based

File: docs/development.rst [hunk 2]
```diff
@@ -21,7 +21,7 @@ Steps:
 
 #. Ask the `VIB-SingleCell-NF` administrators to create a new repository (in this case: ``harmony``) or create one on your GitHub account that could be brought into the `VIB-SingleCell-NF` organization.
 
-    When using your own repo, you MUST start from the `template repository`_ in the vib-singlecell-nf organisation. Click the green "Use this template" button and provide a name for your new repo.
+    When using your own repo, you MUST start from the `template repository`_ in the vib-singlecell-nf organisation. Click the green "Use this template" button and provide a name for your new repo. Make sur the "Include all branches" checkbox is checked.
```
Comment (KrisDavie) [inline docs/development.rst:24]:
```suggestion
    When using your own repo, you MUST start from the `template repository`_ in the vib-singlecell-nf organisation. Click the green "Use this template" button and provide a name for your new repo. Make sure the "Include all branches" checkbox is checked.
```

File: docs/features.rst [hunk 3]
```diff
@@ -55,7 +55,7 @@ Finally run the pipeline,
 
 Set the seed
 ------------
-Some steps in the pipelines are nondeterministic. To be able that the results are reproducible in time, by default a seed is set to:
+Some steps in the pipelines are nondeterministic. In order to have reproducible results in time, a seed is set by default to:
```
Comment (KrisDavie) [inline docs/features.rst:58]:
```suggestion
Some steps in the pipelines are nondeterministic. In order to have reproducible results, a seed is set by default to:
```

File: docs/features.rst [hunk 4]
```diff
@@ -117,20 +117,40 @@ The profile ``utils_cell_annotate`` should be added when generating the main con
...
 
+- ``off`` should be set to ``h5ad``
+- ``method`` choose either ``obo`` or ``aio``
 - ``cellMetaDataFilePath`` is a TSV file (with header) with at least 2 columns: a column containing all the cell IDs and an annotation column.
-- ``indexColumnName`` is the column name from ``cellMetaDataFilePath`` containing the cell IDs information.
+
+If ``obo`` is used, the following params are required:
+
+- ``sampleSuffixWithExtension`` is the suffix used to extract the sample ID from the file name of ``cellMetaDataFilePath`` i.e.: it is the suffix after the sample name in the file path.
+- ``indexColumnName`` is the column name from ``cellMetaDataFilePath`` containing the cell IDs information. This column **should** have unique values.
```
Comment (KrisDavie) [inline docs/features.rst:147]: Is this should have unique, or must have unique?

File: workflows/single_sample.nf [hunk 5]
```diff
@@ -3,24 +3,13 @@ nextflow.preview.dsl=2
...
-include CLUSTER_IDENTIFICATION from '../src/scanpy/workflows/cluster_identification.nf' params(params)
 include SC__H5AD_TO_FILTERED_LOOM from '../src/utils/processes/h5adToLoom.nf' params(params)
 include FILE_CONVERTER from '../src/utils/workflows/fileConverter.nf' params(params)
 
-// reporting:
-include UTILS__GENERATE_WORKFLOW_CONFIG_REPORT from '../src/utils/processes/reports.nf' params(params)
-include SC__SCANPY__MERGE_REPORTS from '../src/scanpy/processes/reports.nf' params(params)
-include SC__SCANPY__REPORT_TO_HTML from '../src/scanpy/processes/reports.nf' params(params)
+// Pipeline
+include single_sample as SCANPY_SINGLE_SAMPLE from '../src/scanpy/main.nf' params(params)
```
Comment (KrisDavie) [inline workflows/single_sample.nf:12]: I really don't like the fact that Scanpy now depends on the scrublet module. I thought we would move towards making the modules independent from each other and deal with links through workflows on the main repo?

File: docs/features.rst (hunk 4 above)
Comment (dweemx) [reply docs/features.rst:147]: `must` is a better word :+1:

File: workflows/single_sample.nf (hunk 5 above)
Comment (dweemx) [reply workflows/single_sample.nf:12]: Ok so you would move the `single_sample_scrublet` pipeline in the `vsn-pipelines` ?
The thing is `scanpy` is also dependent on `pcacv` so it's tricky

**Comment by cflerin** [comment]
I agree that having sub-modules be dependent on each other has the potential to get a bit messy (although I'm not really sure of the cases where we would need to clone and run the Scanpy module on its own, for instance).

Ideally we would have the complex workflows that depend on multiple tools in the main repo `workflows` directory (`single_sample`, etc.). There's nothing about `single_sample` that make it necessary to put it in the scanpy module, right?

Then, within the sub-modules (scanpy, etc.) we would have sub-workflows that use only tools from that module?

**Comment by dweemx** [comment]
> I agree that having sub-modules be dependent on each other has the potential to get a bit messy (although I'm not really sure of the cases where we would need to clone and run the Scanpy module on its own, for instance).
> 
> Ideally we would have the complex workflows that depend on multiple tools in the main repo `workflows` directory (`single_sample`, etc.). There's nothing about `single_sample` that make it necessary to put it in the scanpy module, right?
> 
> Then, within the sub-modules (scanpy, etc.) we would have sub-workflows that use only tools from that module?

Ok you'd put the `single_sample_scrublet` pipeline as a main workflow in `vsn-pipelines` instead of `scanpy` then ?

**Comment by cflerin** [comment]
> Ok you'd put the `single_sample_scrublet` pipeline as a main workflow in `vsn-pipelines` instead of `scanpy` then ?

Yes, I think it makes the most sense. This is currently where the "meta" workflows are (`single_sample`, `bbknn`, etc.) and it seems like `single_sample_scrublet` fits into that category (correct me if I'm wrong).

**Comment by KrisDavie** [comment]
> Ideally we would have the complex workflows that depend on multiple tools in the main repo `workflows` directory (`single_sample`, etc.). There's nothing about `single_sample` that make it necessary to put it in the scanpy module, right?
> 
> Then, within the sub-modules (scanpy, etc.) we would have sub-workflows that use only tools from that module?

This I completely agree with, this is how I envisioned it working. This also allows other people to use our modules in their own repos/workflows without needed the entire vsn-pipelines

> Ok you'd put the single_sample_scrublet pipeline as a main workflow in vsn-pipelines instead of scanpy then ?

Yeah, this makes the most sense to me.

File: workflows/single_sample.nf (hunk 5 above)
Comment (dweemx) [reply workflows/single_sample.nf]: Removed `single_sample_scrublet` from `scanpy` repo to `vsn-pipelines`

**Pull request merged**